torch/nn/functional.py

```diff
@@ -1308,6 +1318,73 @@ def layer_norm(input, normalized_shape, running_mean, running_var,
         return out
 
 
+def group_norm(input, num_groups, running_mean=None, running_var=None,
+               weight=None, bias=None, use_input_stats=True,
+               momentum=0.1, eps=1e-5):
+    r"""Applies Group Normalization for last certain number of dimensions.
+
+    See :class:`~torch.nn.GroupNorm` for details.
+    """
+    if not use_input_stats and (running_mean is None or running_var is None):
+        raise ValueError('Expected running_mean and running_var to be not None when use_input_stats=False')
+
+    input_shape = input.size()
+    b = input_shape[0]
+    c = input_shape[1]
+    g = num_groups
+
+    if c % num_groups != 0:
+        raise ValueError('Expected number of channels in input to be divisible '
+                         'by num_groups, but got {} input and num_groups={}'
+                         .format(input_shape, num_groups))
+
+    if weight is not None and (weight.dim() != 1 or weight.numel() != c):
+        raise ValueError('Expected weight to be a vector of size equal to the '
+                         'number of channels in input, but got {} weight and {} '
+                         'input'.format(weight.size(), input_shape))
+
+    if bias is not None and (bias.dim() != 1 or bias.numel() != c):
+        raise ValueError('Expected bias to be a vector of size equal to the '
+                         'number of channels in input, but got {} bias and {} '
+                         'input'.format(bias.size(), input_shape))
+
+    # Repeat stored stats if necessary
+    if running_mean is not None:
+        running_mean_orig = running_mean
+        running_mean = running_mean_orig.repeat(b)
+    if running_var is not None:
+        running_var_orig = running_var
+        running_var = running_var_orig.repeat(b)
+
+    # Apply group norm
+    input_reshaped = input.contiguous().view(1, b * g, -1)
+
```

Comment:
Basically we make BN kernels to normalize the `[c // g, *]` part of `[b, g, c // g, *]`